version: '3'

services:
  rinha_api1_py: &rinha_api1_py
    build: .
    container_name: rinha_api1_py
    network_mode: host
    entrypoint: sanic main:app --port 3001
    restart: unless-stopped
    depends_on:
      postgres_py:
        condition: service_healthy
    deploy:
      resources:
        limits:
          cpus: "0.3"
          memory: "150MB"

  rinha_api2_py:
    <<: *rinha_api1_py
    container_name: rinha_api2_py
    entrypoint: sanic main:app --port 3002

  postgres_py:
    image: postgres:latest
    container_name: rinha_postgres_py
    command: postgres -c config_file=/docker-entrypoint-initdb.d/postgresql.conf
    restart: unless-stopped
    network_mode: host
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 5s
      timeout: 5s
      retries: 20
      start_period: 10s
    environment:
      POSTGRES_DB: mydatabase
      POSTGRES_USER: myuser
      POSTGRES_PASSWORD: mypassword
    volumes:
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql      
      - ./postgresql.conf:/docker-entrypoint-initdb.d/postgresql.conf
    deploy:
      resources:
        limits:
          cpus: '0.6'
          memory: '240MB'

  nginx:
    image: nginx:latest
    container_name: rinha_nginx_py
    network_mode: host
    restart: unless-stopped
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
    depends_on:
      - rinha_api1_py
      - rinha_api2_py
    deploy:
      resources:
        limits:
          cpus: "0.3"
          memory: "10MB" 
